### ISSUE TITLE
Add URL rewrites

### ISSUE THREAD

**Issue by FabianLars** [meilisearch/docs-scraper#378]
**Description**
My use case would be to scrape my website on localhost before it's live and have the scraper replace the localhost address with the known remote url. To be specific: my use case is to scrape the website (on localhost) in a github action while netlify is still processing the git repo.

**Basic example**
Hmm, probably something like this
```json
"rewrite_urls": [
  { "replace": "http://localhost:3000", "with": "https://my-remote-url.com" }
]
```

**Other**

I am somewhat interested to implement this myself if nobody else has time for this, but i don't have much experience with Scrapy and this repo so i won't make any promises.

Oh and of course feel free to close this issue if the request is out of scope for this project :)

**Comment by alallema** [comment]
Hi @FabianLars,
Sorry for the delay 😅 ! I'm not sure I understand precisely what you want to implement. Do you want to add a feature to this repo? Could you be a little more specific?
From what I understand I don't think this is part of the scope of this repo which is intended to be a scrapper there is no usage of netlify in it

**Comment by FabianLars** [comment]
It basically comes down to this:
> To be specific: my use case is to scrape the website (on localhost) in a github action while netlify is still processing the git repo.

Our website's code is hosted on github, each new commit will be picked up by netlify automatically which in turn builds and deploys the site.
What i'd like to do is scrape the new content _at the same time_ in a normal github action workflow. To do this i'll have to spawn a localhost server (which is probably also a bit faster than a remote server) but that will make all the indexed documents have the localhost address instead of the live version that's still being deployed.

It's not the end of the world if this is not possible, or out of scope btw.

The alternatives we could think of aren't _that_ nice though because we can't trigger a github action from a netlify build, so we have to do one of these 2 options, both first deploy the site and then update the index:
- Deploy to netlify from github action - that's what we did in the past but we've had issues with it i think.
- Start a github action on each commit, and estimate netlify's build time by sleeping, that's what we do right now: <https://github.com/tauri-apps/tauri-docs/blob/beta/.github/workflows/update-beta-index.yml#L32> - this is of course either a bit risky, or wasteful depending on the build times.

tldr: i want this
![grafik](https://github.com/meilisearch/docs-scraper/assets/30730186/32ab6986-759b-4368-8f5b-b59ce4a61ff1)
to show `https://beta.tauri.app` instead without having to scrape beta.tauri.app directly.

**Comment by alallema** [comment]
No problem, but apart from the fact that it is a very specific need I don't see how it could be added in this repo.
However, if you need to scrap a website on localhost I recommend you to use this two parameters:
https://github.com/meilisearch/docs-scraper#js_render-optional
https://github.com/meilisearch/docs-scraper#js_wait-optional
